electron/app/protocol-manager.ts

```diff
@@ -0,0 +1,135 @@
+// protocol.utils.ts
+import { app, ipcMain, powerMonitor } from 'electron';
+import { settingsService } from './services/settings-service';
+import { appConstants } from './app-constants';
+import WindowManager from './window-manager';
+const { Deeplink } = require('electron-deeplink');
+
+const deeplink = new Deeplink({ app })
+
+function isNil(value) {
+    return value == null
+  }
+
+export abstract class ProtocolUtils {
+    // public static setDefaultProtocolClient(): void {
+    //     if (!app.isDefaultProtocolClient(appConstants.PROTOCOL_NAME)) {
+    //         // Define custom protocol handler.
+    //         // Deep linking works on packaged versions of the application!
+    //         app.setAsDefaultProtocolClient(appConstants.PROTOCOL_NAME);
+    //     }
+    // }
+
+    /**
+     * @description Create logic (WIN32 and Linux) for open url from protocol
+     */
+    public static setProtocolHandlerWindowsLinux(): void {
+        // Force Single Instance Application
+        // const gotTheLock = app.requestSingleInstanceLock();
+
+        // app.on('second-instance', (e: Electron.Event, argv: string[]) => {
+        //     // Someone tried to run a second instance, we should focus our window.
+        //     if (WindowManager.mainWindow) {
+        //         if (WindowManager.mainWindow.isMinimized()) WindowManager.mainWindow.restore();
+        //         WindowManager.mainWindow.focus();
+        //     } else {
+        //         // Open main windows
+        //         WindowManager.openMainWindow();
+        //     }
+
+        //     app.whenReady().then(() => {
+        //         WindowManager.mainWindow.loadURL(this._getDeepLinkUrl(argv));
+        //     });
+        // });
+
+        // if (gotTheLock) {
+        //     app.whenReady().then(() => {
+        //         // Open main windows
+        //         WindowManager.openMainWindow();
+        //         WindowManager.mainWindow.loadURL(this._getDeepLinkUrl());
+        //     });
+        // } else {
+        //     app.quit();
+        // }
+        app.whenReady().then(() => {
+            // Open main windows
+            console.log('whenReady');
+            WindowManager.openMainWindow();
+            WindowManager.mainWindow.loadURL(this._getDeepLinkUrl());
+        });
```

Comment:
replace this with the callback.

Reply:
Something like this:
```suggestion
            const rawUrl = this._getDeepLinkUrl();
            deeplinkCallback(rawUrl);
        });
```